architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,227 @@
+  * It is critical that this path preserves the Service IP, if the traffic was to a Service.
+* All traffic *ingressing* a pod on port 15008 in the mesh should be redirected to the node-local ztunnel on port 15008 (more on this later).
+* All other traffic *ingressing* a pod in the mesh should be redirected to the node-local ztunnel on port 15006.
+
+TODO: fill in implementation details of how redirection is actually implemented.
+
+## HBONE
+
+Along with pass-through traffic, Ztunnel supports the "HBONE" protocol.
+This is not really so much a new protocol, but rather a name we came up with to refer to the expectations of clients and servers communicating in the mesh.
+
+HBONE is just a standard HTTP `CONNECT` tunnel, over mutual TLS with mesh (SPIFFE) certificates, on a well known port (15008).
+The target destination address is set in the `:authority` header, and additional headers can be included as well.
+Currently, only HTTP/2 is supported, though HTTP/1.1 and HTTP/3 are planned.
+
+Currently, SNI is not set by Istio clients and ignored by Istio servers.
+This makes identifying which certificate to use problematic for Ztunnel.
+To handle this, requests to Ztunnel are sent to `DestinationPod:15008` and redirected to ztunnel, rather than `ZtunnelPod:15008`.
+The original destination is then extracted to determined which certificate to use.
+SNI is not used because it is illegal to use IP's in SNI, and there is no other existing standard format to represent what we need to.
+Additionally, using the redirection mechanism reduces the need for clients to know the destination's ztunnel address.
+
+### Pooling
+
+User connections can be multiplexed over shared HBONE connections.
+This is done through standard HTTP/2 pooling.
+The pooling is keyed off the `{source identity, destination identity, destination ip}`.
+
+### Headers
+
+Ztunnel uses the following well-known headers in HBONE:
+
+| Header        | Purpose                                                                                                                                                                                                                                                 |
+|---------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `:authority`  | Required in `CONNECT`, this is the target destination                                                                                                                                                                                                   |
+| `Forwarded`   | For outgoing requests, the original source IP. Note that since we spoof IPs in most cases, this usually is the same as the actual IP seen. For incoming requests, this is used only for traffic from waypoints (which are trusted and cannot spoof IPs) |
```

Comment:
The absence of an `X-` prefix is meant to indicate that these headers are standard for HBONE?

Reply:
The lack of X- prefix is just using the industry recommended best practices for headers (X- is not supposed to really be used, especially when there is a standard alternative)